### ISSUE TITLE
[REQUEST] Tractor beam power

### ISSUE THREAD

**Issue by ebsmarch** [Liam-Harrison/Celeritas#305]
When the tractor beam gets closer to the cursor, it should increase in power.

**Comment by MistaMishMash** [comment]
Task Estimation (time): 4 hours
Does this issue address a risk? yes, players find the tractor beam to be difficult and unintuitive to use.

Risks: I don't have a clear idea of what is required to complete this task. It's quite a subjective end-goal.
Risk Resolution: will communicate my progress clearly with the team in the form of gameplay videos, so I can get continuous feedback, and hopefully get to a result that everyone is happy with.

**Comment by MistaMishMash** [comment]
Progress as of Wednesday, have sent to team for feedback. Have added Drag and extra power to the beam the closer it is to the tractor-ed object, scaling w an animation curve.
https://user-images.githubusercontent.com/19755210/135217913-a730a580-8534-4679-9083-2e55e697e0ce.mp4

**Comment by MistaMishMash** [comment]
Have implemented this by adding and exposing some variables that designers can use to change how powerful and floaty the tractor beam feels to use:
![image](https://user-images.githubusercontent.com/19755210/135386194-ef024e01-220f-4133-a42a-2d3cb35c8e84.png)

**Comment by MistaMishMash** [comment]
Actual time taken: 2 hours 20 minutes

Testing procedure: Playtest. Also recorded tractor beam behaviour and sent to team for feedback/approval
Test Pass Condition(s):  Entered the game, threw some rocks around, sent a video to the team for any feedback/things to improve. On team approval, task was completed.

For the wave weapon, first I reproduced the error (shot the weapon, entered a dialogue, saw how the bullets continued to move)
After implementing the fix, I repeated this process to ensure that the bullets stayed stationary.

For the icons, I simply put the modules into my ship, made sure the icons looked OK.

Risks: The tractor beam is a bit of an odd mechanic, players may find it difficult to use at first
Risk Resolution: have added friction so it is less drifty. Designers can also tweak these values further through the inspector. Happy to do further work on this feature if required.
